### ISSUE TITLE
Be more consistent in warning about unnecessary ${} with associative arrays in $((..))

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#1481]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2004
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2004)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

This is a strange case that ShellCheck actually gets right, but for the wrong reasons. `${..}` *is* required for associative arrays with weird key names:

```sh
#!/bin/bash
declare -A arr
x="]"
echo "$(( ${arr[$x]} ))"     # Works
echo "$(( ${arr[bla]} ))"    # Says to remove the ${..}
echo "$(( arr[$x] ))"        # Fails
```

#### Here's what shellcheck currently says:

In foo line 5:
echo "$(( ${arr[x]} ))"
          ^-------^ SC2004: $/${} is unnecessary on arithmetic variables.

#### Here's what I wanted or expected to see:

Probably nothing, since it would be better for consistency if all instances of using arrays were warned about the same way.
